src/server/service/kv.rs

```diff
@@ -823,8 +829,24 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
                         raft_resps
                     ));
                 } else {
-                    let read_index = raft_resps[0].get_read_index().get_read_index();
-                    resp.set_read_index(read_index);
+                    // Update max_read_ts and check memory locks after read index command finishes.
+                    // It reduces the chance of encountering locks if we check it as late as possibile.
+                    match check_memory_locks_fut.await {
+                        Ok(Some(lock)) => {
+                            resp.set_locked(lock);
```

Comment:
Whether leader will check locks in memory after its apply index reach latest commit index? Learner/Follower may apply faster than leader.

Reply:
It needn't.
If we wait until apply, the only effect is that some lock may be not found in the memory lock of the leader, but such a lock must be found in the storage at its applied state. It shouldn't have any difference, I think.

Reply:
Should we do some benchmark before or after merging this RP?